fakebook/src/app/component/profile-view/profile-view.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router'; // getting the id number
+import { User } from 'src/app/model/user';
+import { ProfileService } from '../../service/profile.service';
+
+@Component({
+  selector: 'app-profile-view',
+  providers: [ProfileService],
+  templateUrl: './profile-view.component.html',
+  styleUrls: ['./profile-view.component.css']
+})
+export class ProfileViewComponent implements OnInit {
+  user: User | null = null;
+
+  constructor(
+    private route: ActivatedRoute, // getting the id # in route
+    private profileService: ProfileService
+  ) {}
+
+  ngOnInit(): void {
+    this.getUser();
+  }
+  getUser(): void {
+    let tempId = ''; // the only way i could declare a variable that may accept a null value in thefuture
```

Comment:
this odd workaround isnt needed, just call `get`, assign the return value to a const, then condition on the value of that variable. typescript can't assume that if you call `get` a second time on line 27 it won't be null just because it wasn't null on line 26.